src/sagemaker/workflow/automl_step.py

```diff
@@ -114,7 +115,12 @@ def arguments(self) -> RequestType:
             request_dict.pop("ModelDeployConfig", None)
         if "GenerateCandidateDefinitionsOnly" in request_dict:
             request_dict.pop("GenerateCandidateDefinitionsOnly", None)
-        request_dict.pop("AutoMLJobName", None)
+        if not _pipeline_config or not _pipeline_config.use_custom_job_prefix:
```

Comment:
It looks like this logic block is repeated in a few places, could we extract this into a helper function that takes job_key and other needed parameters?

Reply:
Yep this is a simple quality of life fix. I will do that.